internal/pkg/dir/dir_test.go

```diff
@@ -57,3 +59,56 @@ func TestInit(t *testing.T) {
 	}
 	checkPermissions(t, bar, 0755)
 }
+
+func TestCopy(t *testing.T) {
+	src := t.TempDir()
+	srcDirname := path.Base(src)
+
+	// Test that empty directories don't copy anything
+	dst := t.TempDir()
+	err := Copy(src, dst)
+	if err != nil {
+		t.Fatalf("Unable to copy empty dir: %v", err)
+	}
+
+	//rmdir will fail if the directory has anything at all
+	err = exec.Command("rmdir", path.Join(dst, srcDirname)).Run()
+	if err != nil {
+		t.Fatalf("Unable to remove supposedly empty dir: %v", err)
+	}
+
+	// Test that copying actually works
```

Comment:
To add a bit more structure besides the comments, you could put the different tests either into their own test functions à la `TestCopy_EmptyDirs` and `TestCopy_FilesAndDirs` (or similar), or run them as subtests using `t.Run("EmptyDirs", func (t *testing.T) { ... })`.